src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -139,15 +141,27 @@ public Integer getTotalUnReviewedDars(AuthUser authUser) {
     /**
      * Filter DataAccessRequestManage objects on user.
      *
-     * @param authUser Filter on what DACs the auth user has access to.
+     * @param user Filter on what DARs the user has access to.
      * @return List of DataAccessRequestManage objects
      */
-    public List<DataAccessRequestManage> describeDataAccessRequestManageV2(AuthUser authUser) {
+    public List<DataAccessRequestManage> describeDataAccessRequestManageV2(User user, String roleName) {
+        if (Objects.nonNull(roleName) && Objects.nonNull(user)) {
+            if (roleName.equalsIgnoreCase(UserRoles.SIGNINGOFFICIAL.getRoleName()) && user.hasUserRole( UserRoles.SIGNINGOFFICIAL)) {
+                if (Objects.nonNull(user.getInstitutionId())) {
+                    List<DataAccessRequest> dars = dataAccessRequestDAO.findAllDataAccessRequestsForInstitution(user.getInstitutionId());
+                    List<DataAccessRequest> openDars = filterOutCanceledDars(dars);
+                    return createAccessRequestManageV2(openDars);
+                } else {
+                    throw new NotFoundException("Signing Official (user: " + user.getDisplayName() + ") "
+                      + "is not associated with an Institution.");
+                }
+            } else {
+                throw new ForbiddenException("User: " + user.getDisplayName() + ", " + user.getDacUserId() + " does not have Signing Official role.");
```

Comment:
I'm getting this exception if I try to use other roles, ie. `Member`, `Chairperson`, etc., which is confusing since I'm not asking for SO dars. I think in an earlier comment, I suggested handling the role validation in the resource method which should help avoid putting too much logic into the service method. If we validated that the user has the role they are asking for there, then here in the service class, we would only need to check the roleName === SO case and then continue on with the rest of the method in all other cases.

Reply:
Also, I'm still seeing the `user.getDacUserId()` in the error message which was another previous PR comment.

Reply:
FYI, I also see this error when I try a non-valid role, like `french` or `fries`. We definitely need some input validation in the resource class to prevent this kind of error.

Reply:
Okay I just moved the SO role validation to the resource class, and added roleName validation in the resource class, as well as corresponding tests